### ISSUE TITLE
confusing with pretrained model finetune tutorial

### ISSUE THREAD

**Issue by Godricly** [zackchase/mxnet-the-straight-dope#178]
In the following part of code.
```python
deep_dog_net = models.squeezenet1_1(prefix='deep_dog_', classes=2)
deep_dog_net.collect_params().initialize()
deep_dog_net._children[0] = net.features
print(deep_dog_net)
```
What is the _children here? It's a little confusing.
what if i use?
```
deep_dog_net.features = net.features
```
@zhreshold

**Comment by zackchase** [comment]
Paging primary contributor for this chapter: @szha can you take this one?

**Comment by szha** [comment]
@Godricly The way you described should be the right way to do it.

**Comment by szha** [comment]
I'm making a PR to clean up usage of hacks like this.

**Comment by szha** [comment]
There will be some code changes needed in mxnet first.

**Comment by zhreshold** [comment]
Good point.

**Comment by Godricly** [comment]
Can you check this code please? 
The length of _children is increased by one using ```doc_net.features = net.features```. I haven't finished my code so I'm not sure if this could a issue. But it looks like that I append sth here other than replacing the one I want.

```python
net = models.inception_v3(pretrained=True)
doc_net = models.Inception3(classes=2)
print(len(doc_net._children))
doc_net._children[0] = net.features
print(len(doc_net._children))
doc_net.features = net.features
print(len(doc_net._children))
```

**Comment by szha** [comment]
This behavior is being fixed in apache/incubator-mxnet#7667, and after that change is merged, you should be able to simply change the `net.features` attribute.

**Comment by Godricly** [comment]
cool. Thank you.
